tests/test_arrow_dataset.py

```diff
@@ -750,9 +750,11 @@ def test_flatten(self, in_memory):
             ) as dset:
                 with self._to(in_memory, tmp_dir, dset) as dset:
                     fingerprint = dset._fingerprint
+                    dset._format_columns = ["a.b.c", "foo"]
```

Comment:
I think it should be "a" - "a.b.c" is not a column yet at this point
```suggestion
                    dset._format_columns = ["a", "foo"]
```

Reply:
@lhoestq, you're right, that column is created on `dset.flatten()`, I've tested it and it works, so I'll commit your suggestion! :hugs: